### PR TITLE
Resolve #10949: use raise from in json_format.py

### LINKED ISSUE #10949: Use `raise ... from e` instead of `raise ...` only when catching error with try-except in python
**What language does this apply to?**
Python

**Describe the problem you are trying to solve.**
In some of the python code, we only `raise` an error without using the `from` after catching an error with `except`.
For example: [json_format.py ](https://github.com/protocolbuffers/protobuf/blob/9b81aaae58be0a4db4c06180042e035c383c7e90/python/google/protobuf/json_format.py#L627-L635)
This makes debugging sometimes difficult, especially the try block is a huge chunk of logic. We are not sure which line of code caused the error.

**Describe the solution you'd like**
Use `raise ... from ...` instead of `raise ...` only

**Describe alternatives you've considered**

**Additional context**
Happy to work on this if we think it's a good idea.

### PR DESCRIPTION
**Pull request by TommyDew42**
Resolve #10949

### CODE REVIEW

**Comment by TommyDew42** [comment]
I noticed we don't have python unit tests. Why's that?

**Comment by zhangskz** [comment]
You might be looking for https://github.com/protocolbuffers/protobuf/blob/main/python/google/protobuf/internal/json_format_test.py?

In general you can find python unit tests: https://github.com/protocolbuffers/protobuf/blob/2880fef06cb7443ba24dc1264ba9f02115407f2c/python/BUILD.bazel#L158

**Review by zhangskz — APPROVED** [review]

File: python/google/protobuf/json_format.py
Comment (jorgbrown) [inline python/google/protobuf/json_format.py:404]: This line is more than 80 columns side; please shrink it.  I suggest:

+   raise TypeError('Can not find message descriptor by type_url: {0}'.format(
+       type_url)) from e

File: python/google/protobuf/json_format.py
Comment (jorgbrown) [inline python/google/protobuf/json_format.py:629]: This line is more than 80 columns side; please shrink it.  I suggest a line-break after the comma:
```
+          raise ParseError('Failed to parse {0} field: {1}.'.format(name,
+                                                                    e)) from e
```

Same for lines 633 and 635

File: python/google/protobuf/json_format.py
Comment (jorgbrown) [inline python/google/protobuf/json_format.py:645]: This line is more than 80 columns wide; please shrink it. I suggest a line-break before path:
```
+          '@type is missing when parsing any message at {0}'.format(
+              path)) from e
```

Comment (TommyDew42) [reply python/google/protobuf/json_format.py]: 3087c139cb0d9812724a6417057853720ae4147e & 571893514cc27df5557de8fae93b66036205f671!

**Comment by TommyDew42** [comment]
Seems the `Linux Ruby Release` CI check failed. It doesn't seem we have changes related to ruby in this PR. Can we rerun that again?

**Comment by TommyDew42** [comment]
There isn't much information in the [error](https://source.cloud.google.com/results/invocations/8e3f693f-4bfa-40a1-8c6b-630a14433cf9/targets) page. Would the maintainers have more insight into the CI errors?

I thought the feature branch might be missing sth from the main so it caused the CI errors but my merging failed. How should we merge the master into the feature branch? 😢

**Comment by zhangskz** [comment]
> There isn't much information in the [error](https://source.cloud.google.com/results/invocations/8e3f693f-4bfa-40a1-8c6b-630a14433cf9/targets) page. Would the maintainers have more insight into the CI errors?
> 
> I thought the feature branch might be missing sth from the main so it caused the CI errors but my merging failed. How should we merge the master into the feature branch? 😢

I believe the ruby error was failing on main and should be addressed now. Rerunning Kokoro, but I believe you may need to rebase your PR.

**Comment by TommyDew42** [comment]
I did a rebase. The `Mergeable` check is still failing. Is it because we have a merge commit? Should I erase that in the commit history?

**Comment by zhangskz** [comment]
> I did a rebase. The `Mergeable` check is still failing. Is it because we have a merge commit? Should I erase that in the commit history?
> 
> And out of curiosity, why is merge not encouraged?

You can ignore the Mergeable check -- that failure should always be present to remind Protobuf maintainers to not merge PRs directly and let our automated Copybara system handle merging changes in Github. This is because we land our changes internally first.

**Comment by TommyDew42** [comment]
Do we need one more approval to unblock merging this PR? Tagging @jorgbrown to help!!

**Comment by TommyDew42** [comment]
Would @acozzette be able to help review? All relevant changes in the PR is in python/google/protobuf/json_format.py!!

**Comment by TommyDew42** [comment]
Undid the rebase by git reset and force push to make the PR reviewable again

**Review by acozzette — APPROVED** [review]

**Pull request closed without merging**
